### DISCUSSION TITLE
Events and other resources not deleted in OKD 4.13

### DISCUSSION THREAD

**Discussion by joseluissmarting** [okd-project/okd#1867] [Q&A]
Hi all,

OpenShift version
    4.13.0-0.okd-2023-06-24-145750

Problem:
- The events in all namespaces are not being deleted, there are events since 1 month ago and more. Even if in the ConfigMap config is: "event-ttl":["3h"]" in the apiserver. It should only store the 3h events.
![image](https://github.com/okd-project/okd/assets/130544456/edec0717-18bf-4cc3-a9b6-4447e0c73829)

- The config maps and secrets in openshift-kube-apiserver created for every update, don't rotate either, I think it should store only the last 10 updates, and it doesn't delete the previous ones, I am in the 66:
![image](https://github.com/okd-project/okd/assets/130544456/5cea4bbb-d71c-4a78-9092-cce37dc2a9dd)
And the old ones are still there:
![image](https://github.com/okd-project/okd/assets/130544456/1d802514-0c4b-4ce3-bd79-8a7eac0e86d3)

Consequences:
Last week I get the etcd cluster full because of the thousands of objects in the cluster, events mostly (alarm:NOSPACE). I am deleting then right now manually.

Solution:
How can I rotate the events and resources correctly??

Kind regards

**Comment by vrutkovs** [comment]
>Even if in the ConfigMap config is: "event-ttl":["3h"]" in the apiserver.

That is not supported, I'm not sure what configmap you mean

>it doesn't delete the previous ones

Is it for `bound-sa-token-signing` or you also have all "installer" pods?

**Reply by joseluissmarting** [reply]
Yes, I know it's not supported to change that value, 3h it's the default value. So it's suppose to store only the 3h events, not more

And yes, I have all the "installer" configmaps and secrets, although I already delete some of them.

**Reply by joseluissmarting** [reply]
Any idea why it doesn't delete the old events?
maybe something with okd 4.13 ?
I have some pods that are restarting all the time, and they create a lot of events, maybe they full the etcd and then it started to work wrong ? 
I had over 500.000 events

**Reply by vrutkovs** [reply]
We'd need must-gather to find out more - it could be one of known issues e.g. https://issues.redhat.com/browse/API-1650

**Comment by vrutkovs** [comment]
As for "what's eating my etcd space" see https://etcd.io/blog/2023/how_to_debug_large_db_size_issue/#what-data-occupies-most-of-the-storage-space - it should give you a list of objects. Later you can figure out which namespaces are these coming from, if its indeed from `openshift-*` then its a bug
